### PR TITLE
Start using the `block.json` metadata file for `Products by Tag`

### PR DESCRIPTION
**Pull request by albarin**
The goal of this PR is to use the `block.json` metadata file to register the `Products by Tag` block.

### Testing

How to test the changes in this Pull Request:
1. Create a page and add a `Products by Tag` block.
2. Save the page and check it renders correctly the default config in the frontend.
3. Edit the page again and make some changes to the block (hiding some content, changing columns number, ordering, etc.).
4. Save the page and check it renders properly with the new config.

### CODE REVIEW

**Review by frontdevde — CHANGES_REQUESTED** [review]
When comparing the result in this branch to `trunk` the icon shape and color appear to change. Is it safe to assume we'd want the icon to stay consistent with all other WooCommerce Blocks icons?

|Trunk|Branch|
|-|-|
|Screenshot 2022-05-19 at 10 22 11|Screenshot 2022-05-19 at 10 15 53|

File: assets/js/blocks/handpicked-products/block.json [hunk 1]
```diff
@@ -28,6 +28,13 @@
         },
         "contentVisibility": {
             "type": "object",
+            "default": {
```
Comment (frontdevde) [inline assets/js/blocks/handpicked-products/block.json:31]: Just double-checking, are changes to the `Handpicked Products` block intended in this `Products by Tag` PR?

Comment (albarin) [reply assets/js/blocks/handpicked-products/block.json:31]: @frontdevde I just found that was missing while doing this PR and included here the change 😅

**Comment by albarin** [comment]
> When comparing the result in this branch to `trunk` the icon shape and color appear to change. Is it safe to assume we'd want the icon to stay consistent with all other WooCommerce Blocks icons?

Good catch! Just pushed a fixed for it, thanks! 😊

**Review by frontdevde — APPROVED** [review]
Thanks for promptly addressing the feedback!
I've tested extensively (comparing against `trunk`) and from what I can tell all behavior appears to be as expected. Thanks for working on this, lgtm!

**Pull request merged**